File: Python/Hewan.py
from abc import ABC, abstractmethod
from Pemilik import Pemilik

class Hewan(ABC):
    """Kelas abstrak dasar untuk semua jenis hewan. (Base Class)"""
    def __init__(self, nama: str = "", umur: int = 0, berat: int = 0, pemilik: Pemilik = None):
        self.__nama = nama
        self.__umur = umur
        self.__berat = berat
        self.__pemilik = pemilik 

    # Getter konvensional
    def get_nama(self) -> str:
        return self.__nama

    def get_umur(self) -> int:
        return self.__umur

    def get_berat(self) -> int:
        return self.__berat

    def get_pemilik(self) -> Pemilik:
        return self.__pemilik

    # Setter konvensional
    def set_nama(self, nama: str):
        self.__nama = nama

    def set_umur(self, umur: int):
        if umur >= 0:
            self.__umur = umur

    def set_berat(self, berat: int):
        if berat >= 0:
            self.__berat = berat

    def set_pemilik(self, pemilik: Pemilik):
        self.__pemilik = pemilik

    @abstractmethod
    def get_spesific_info(self) -> str:
        pass

    def get_info_dasar(self) -> str:
        return f"{self.__nama} | Umur: {self.__umur} th | Berat: {self.__berat} kg"

File: Java/Hewan.java
public abstract class Hewan {
    // Atribut private
    private String nama;
    private int umur;
    private int berat;
    // Agregasi: Objek Pemilik
    private Pemilik pemilik;

    // Konstruktor
    public Hewan(String nama, int umur, int berat, Pemilik pemilik) {
        this.nama = nama;
        this.umur = umur;
        this.berat = berat;
        this.pemilik = pemilik;
    }

    // Konstruktor default
    public Hewan() {
        this("", 0, 0, null);
    }

    // Getter konvensional
    public String getNama() {
        return nama;
    }

    public int getUmur() {
        return umur;
    }

    public int getBerat() {
        return berat;
    }

    public Pemilik getPemilik() {
        return pemilik;
    }

    // Setter konvensional
    public void setNama(String nama) {
        this.nama = nama;
    }

    public void setUmur(int umur) {
        if (umur >= 0) {
            this.umur = umur;
        }
    }

    public void setBerat(int berat) {
        if (berat >= 0) {
            this.berat = berat;
        }
    }

    public void setPemilik(Pemilik pemilik) {
        this.pemilik = pemilik;
    }

    public abstract String getSpesificInfo();

    public String getInfoDasar() {
        return nama + " | Umur: " + umur + " th | Berat: " + berat + " kg";
    }
}